timemachine/fe/rbfe.py

```diff
@@ -199,41 +213,73 @@ def setup_initial_states(
     # perform any concluding sanity-checks
     assert_all_states_have_same_masses(initial_states)
 
-    return initial_states
+    def make_optimized_initial_state(lamb: float) -> InitialState:
+        initial_state = make_initial_state(lamb)
+
+        # use pre-optimized initial state with the closest value of lambda as a starting point for optimization
+        nearest_optimized = min(initial_states, key=lambda s: abs(lamb - s.lamb))
+
+        if lamb == nearest_optimized.lamb:
+            x_opt = nearest_optimized.x0
```

Comment:
Should we just return `nearest_optimized`? It seems confusing and perhaps buggy to have multiple initial state objects with the same lambda value.

Reply:
Yes, I think that's better, good call. Fixed in https://github.com/proteneer/timemachine/pull/959/commits/68b17772e3a32a2b6ffd9f2a9a24d1abcb04c388